### ISSUE TITLE
[BUG] Autolearn is enabled for all classifiers if at least one classifier has it enabled

### ISSUE THREAD

**Issue by dragoangel** [rspamd/rspamd#4974]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Tried ASAN package and obtained the ASAN report (if possible): https://rspamd.com/doc/faq.html#asan-builds 
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**
Setting an auto learn setting to false - enables bayes autolearning on classifier, which is very confusion and can result into unexpected results, like overload of system, or messing up bayes statistics database data.

### Steps to Reproduce

1. Configure `classifier-bayes.conf` with `autolearn = false;`
2. Start scanning emails
3. Check logs or bayes redis, there will be traces of autolearning

**Expected behavior**
Autolearning is turned off and no statistics created for classifier.

### Versions

3.9.0 from master

### Additional Information

There no docs hot wo disable autolearn without dropping it from configuration, but from what I have tested, next options are valid: `autolearn {};` and `autolearn = [];`, only `false` treated as `true`.

I think error is here: https://github.com/rspamd/rspamd/blob/c13a45b71d7edcb692c2f1269a517bf26d471b3c/src/libstat/stat_process.c#L1001

Looks like it missing condition check that it's `true` or `false`, but mine C skills are -1 to provide even such basic PR, sorry.

**Comment by vstakhov** [comment]
No, it should work as intended.

**Comment by vstakhov** [comment]
I think your issue is somewhere else.

**Comment by dragoangel** [comment]
Hi @vstakhov, you are right! I triple checked, and it really works.

**Comment by dragoangel** [comment]
@vstakhov I found what was the case, actually this happen when enable multiply classifiers and one classifier have autolearning enabled, and other is not, like:
```json
classifier {
  bayes {
    per_user = true;
    autolearn = false;
    expiry = false;
    backend = "redis";
    read_servers = "peruser-master";
    write_servers = "peruser-replica-1,peruser-replica-2";
    db = "0";
    min_tokens = 11;
    min_learns = 1;
    statfile {
        symbol = "BAYES_USER_HAM";
        spam = false;
    }
    statfile {
        symbol = "BAYES_USER_SPAM";
        spam = true;
    }
    learn_condition = 'return require("lua_bayes_learn").can_learn';
    new_schema = true;
    store_tokens = false;
    signatures = false;
    cache_prefix = "learned_ids";
    cache_max_elt = 50000;
    cache_max_keys = 5;
    cache_elt_len = 32;
    cache {}
    tokenizer {
      name = "osb";
    }
  }
}

classifier {
  bayes {
    per_user = false;
    autolearn = [-3, 8];
    expiry = false;
    backend = "redis";
    read_servers = "global-master";
    write_servers = "global-replica-1,global-replica-2";
    db = "0";
    min_tokens = 11;
    min_learns = 200;
    statfile {
        symbol = "BAYES_HAM";
        spam = false;
    }
    statfile {
        symbol = "BAYES_SPAM";
        spam = true;
    }
    learn_condition = 'return require("lua_bayes_learn").can_learn';
    new_schema = true;
    store_tokens = false;
    signatures = false;
    cache_prefix = "learned_ids";
    cache_max_elt = 50000;
    cache_max_keys = 5;
    cache_elt_len = 32;
    cache {}
    tokenizer {
      name = "osb";
    }
  }
}
```

Result: both redises would get learned keys, while expected only `bayes-global` had to autolearn.

**Comment by dragoangel** [comment]
@vstakhov thank you for a help in PM, my config was missing `name` for classifier like:
~~~hcl
classifier "bayes" {
  name = "bayes_peruser";
  per_user = true;
  autolearn = false;
  ...
}

classifier "bayes" {
  name = "bayes_global";
  per_user = false;
  autolearn = [-3, 8];
  ...
}
~~~

Missing `name` result in autolearn treat both bayes as one thing

**Comment by akkyoh** [comment]
Hi all. I encountered a similar problem. The problem is not the lack of a `name` parameter. I had this parameter present in both classifiers.

My configuration file had the settings shown below. The autolearn block was taken from the documentation https://rspamd.com/doc/configuration/statistic.html. Learning happened for user and global simultaneously until I wrote `autolearn = [-0.5, 6]`

```
classifier "bayes" {
 name = "user";
 per_user = true;
 autolearn = false;
 ...
}

classifier "bayes" {
 name = "global";
 per_user = false;
 autolearn {
 spam_threshold = 6;
 ham_threshold = -0.5;
 check_balance = true;
 min_balance = 0.9;
 }
 ...
}
```
